feature Система сохранений

i-shindev opened the pull request:
Реализована иерархическая система сохранений, основанная на Newtonsoft.JSON.

i-shindev committed:
Added hierachy save system. Based on Newtonsoft.JSON

i-shindev committed:
Fixed commentaries

Kukushenok commented:
1) Поправь наименования protected/private переменных
2) Вынеси Newtonsoft.JSON из жёсткой зависимости (создай базовый класс - интерфейс JObject; а после унаследуйся от него и сделай JSONSaveObject)
3) Задокументируй, как можно пользоваться
синглтон, конечно нехорошо, но пускай....

i-shindev committed:
Рефакторинг.
- Теперь это ScriptableObject, а не синглтон
- Папка Features
- Нужно поменять энкодинг поэтому делаю коммит

i-shindev committed:
Добавил фичу - SaveSystem, позволяющая создавать иерархические менеджеры сохранений как ScriptableObject'ы

i-shindev committed:
Крошечный рефакторинг

Kukushenok commented:
так ну я вроде исправил многое надо открыть снова как не драфт

Kukushenok commented:
ладно я как админ могу без проверки мерджнуть хотя нужно чтобы хотя бы один посмотрел.
**Если найдёте непонятности или ошибки - пишите мне чтобы я исправил** :trollface:

Kukushenok commented:
я забыл что коммиты на английском писать надо :skull:

Kukushenok commented:
для моей фичи описание такое:
Представляет собой базу для системы сохранений на основе менеджеров - Scriptable Objects. 
Наследуясь от класса BaseDataSaver, можно реализовать по-своему методы Load и Save.
Если какому-то скрипту в сцене нужно обращаться к какой-то системе сохранения, то тогда он просто ссылается на нужный Scriptable Object.
Сохранения находятся по %AppData%/LocalLow/DefaultCompany/Evercold/saved_game.save и представлены в формате JSON.

Откройте Scenes/SaveSystem и запустите. Изменяйте объекты в ExampleSaveSystem. После этого, остановив проект, снова их поменяйте. Запустив снова, вы убедитесь, что им вернулось старое значение.

Kukushenok commented:
ну и да, ветка после этого удаляется.
если нужно будет пофиксить фичу - нужно будет создать новую ветку.

Kukushenok commented:
а, тут можно restore branch нажать, ну тогда ладно, думаю логичнее ресторить будет :trollface:
